src/services/point.service.ts

```diff
@@ -1,22 +1,61 @@
-import { Guild } from 'discord.js';
+import { Guild, Role, TextChannel, MessageEmbed, EmojiIdentifierResolvable } from 'discord.js';
 import { IUserPoints, Maybe } from '../common/types';
 import { PointsDocument, PointsModel } from '../schemas/points.schema';
 import { GuildService } from './guild.service';
+import Constants from '../common/constants';
+import { UserService } from './user.service';
 
 export class PointService {
   private _guild: Guild;
+  private _userservice: UserService;
+  private _tacoKingRole: Role;
+  private _tacoKingEmoji: Maybe<EmojiIdentifierResolvable> = null;
 
-  constructor(private _guildService: GuildService) {
+  constructor(private _guildService: GuildService, private _userService: UserService) {
     this._guild = this._guildService.get();
+    this._userservice = this._userService;
+    this._tacoKingRole = this._guild.roles.cache.find((r) => r.name === Constants.Roles.TacoKing)!;
+    this._tacoKingEmoji = this._guildService.getEmoji('tacoking') ?? '🌮';
   }
 
   public async awardPoints(id: Maybe<string>, amount: number) {
     if (!id) {
       return;
     }
 
+    // cache the #1 point holder before points are awarded
+    const kingBeforeAwardID = (await this.getTopPoints(1))[0].userID;
+
     const userDoc = await this.getUserPointDoc(id);
     await PointsModel.updateOne(userDoc, { $inc: { numPoints: amount } });
+
+    const kingAfterAwardID = (await this.getTopPoints(1))[0].userID;
+
+    // the king has changed!
+    if (kingBeforeAwardID !== kingAfterAwardID) {
+      const prevKingUser = this._userservice.getMember(kingBeforeAwardID);
+      const newKingUser = this._userservice.getMember(kingAfterAwardID);
+
+      // remove the taco king role from the previous king
+      await prevKingUser?.roles.remove(this._tacoKingRole);
+
+      // give the role to the new king
+      await newKingUser?.roles.add(this._tacoKingRole);
```

Comment:
```suggestion
      // remove the taco king role from the previous king
      // and give the role to the new king
      await Promise.all([
      	prevKingUser?.roles.remove(this._tacoKingRole),
      	newKingUser?.roles.add(this._tacoKingRole),
     ]);
```
This can both run concurrently since one doesn't strictly depend on the other.